src/executor/admin/KillQueryExecutor.cpp

```diff
@@ -0,0 +1,152 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "executor/admin/KillQueryExecutor.h"
+
+#include "context/QueryContext.h"
+#include "planner/plan/Admin.h"
+
+namespace nebula {
+namespace graph {
+folly::Future<Status> KillQueryExecutor::execute() {
+    SCOPED_TIMER(&execTime_);
+
+    // TODO: permision check
+
+    QueriesMap toBeVerifiedQueries;
+    QueriesMap killQueries;
+    NG_RETURN_IF_ERROR(verifyTheQueriesByLocalCache(toBeVerifiedQueries, killQueries));
+
+    folly::Promise<Status> pro;
+    auto result = pro.getFuture();
+    qctx()->getMetaClient()->listSessions().via(runner()).thenValue(
+        [toBeVerifiedQueries = std::move(toBeVerifiedQueries),
+         killQueries = std::move(killQueries),
+         pro = std::move(pro),
+         this](StatusOr<meta::cpp2::ListSessionsResp> listResp) mutable {
+            std::vector<meta::cpp2::Session> sessionsInMeta;
+            if (listResp.ok()) {
+                sessionsInMeta = std::move(listResp.value()).get_sessions();
+            } else {
+                LOG(ERROR) << listResp.status();
+            }
+
+            auto status = verifyTheQueriesByMetaInfo(toBeVerifiedQueries, sessionsInMeta);
+            if (!status.ok()) {
+                pro.setValue(std::move(status));
+                return;
+            }
+
+            killCurrentHostQueries(killQueries);
+
+            // upload all queries to be killed to meta.
+            qctx()
+                ->getMetaClient()
+                ->killQuery(std::move(killQueries))
+                .via(runner())
+                .thenValue([pro = std::move(pro), this](auto&& resp) mutable {
```

Comment:
It's better to chain these async futures, otherwise you must handle the exception threw by `killQuery` interface.

```cpp
auto metaClient = qctx()->getMetaClient();
return getMetaClient->listSession()
  .via(runner())
  .thenValue([metaClient](listResp){
    // ...blabla
    return metaClient->killQuery(killedQueries);
  })
  .thenValue([](auto &&resp){
    // ... some status
    return status;
  });
```